src/MIDebugEngine/AD7.Impl/AD7Property.cs

```diff
@@ -82,13 +82,15 @@ public DEBUG_PROPERTY_INFO ConstructDebugPropertyInfo(enum_DEBUGPROP_INFO_FLAGS
                 } else
                 {
                     propertyInfo.dwAttrib |= enum_DBG_ATTRIB_FLAGS.DBG_ATTRIB_DATA;
-                    string fullName = variable.FullName();
-                    if (_engine.DebuggedProcess.DataBreakpointVariables.Contains(fullName))
+                    try
```

Comment:
You need an optimization here where we only call `variable.Address()` if DataBreakpointVariables has an entry that might match:

1. Move the call to `variable.FullName()` outside of ifs and assign it to a temporary.
2. Here add:

```C#
                
                    if (!string.IsNullOrEmpty(fullName))
                    {
                        lock (_engine.DebuggedProcess.DataBreakpointVariables)
                        {
                            if (_engine.DebuggedProcess.DataBreakpointVariables.Any(candidate =>
                                candidate.Length > fullName.Length
                                && candidate.EndsWith(fullName, StringComparison.Ordinal)
                                && candidate[candidate.Length - 1 - fullName.Length] == ','))
                            {
<your-code-here>
                            }
                        }
                    }
```

Reply:
FYI This is why you were seeing refresh being so slow.

Reply:
Sorry if this is obvious, but you will need to add `using System.Linq;` for `.Any` to work.